src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/XElementProcessor.cs

```diff
@@ -0,0 +1,231 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Exceptions;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Extensions;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Models.KnownStrings;
+using Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.ReferenceRegistries;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Readers;
+
+namespace Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration
+{
+    /// <summary>
+    /// Holds common functionality to process xelement that are used across various <see cref="IFilter"/>.
+    /// </summary>
+    public static class XElementProcessor
```

Comment:
>XElementProcessor [](start = 24, length = 17)

Consider "XElementExtensions" instead and change all of the methods in this class to actual extension methods. This would be the recommended pattern I believe.